packages/alfa-rules/test/common/predicate/is-visible.spec.tsx

```diff
@@ -94,3 +94,229 @@ test("isVisible() returns true for replaced elements with no child", (t) => {
 
   t.equal(isVisible(element), true);
 });
+
+test(`isVisible() returns false for an element that has been pulled offscreen
+      by position: absolute and left: 9999px`, (t) => {
+  const element = (
+    <div
+      style={{
+        position: "absolute",
+        left: "9999px",
+      }}
+    >
+      Hello world
+    </div>
+  );
+
+  t.equal(isVisible(element), false);
+});
+
+test(`isVisible() returns false for an element that has been pulled offscreen
+      by position: absolute and left: -9999px`, (t) => {
+  const element = (
+    <div
+      style={{
+        position: "absolute",
+        left: "-9999px",
+      }}
+    >
+      Hello world
+    </div>
+  );
+
+  t.equal(isVisible(element), false);
+});
+
+test(`isVisible() returns false for an element that has been pulled offscreen
+      by position: absolute and right: 9999px`, (t) => {
+  const element = (
+    <div
+      style={{
+        position: "absolute",
+        right: "9999px",
+      }}
+    >
+      Hello world
+    </div>
+  );
+
+  t.equal(isVisible(element), false);
+});
+
+test(`isVisible() returns false for an element that has been pulled offscreen
+      by position: absolute and right: -9999px`, (t) => {
+  const element = (
+    <div
+      style={{
+        position: "absolute",
+        right: "-9999px",
+      }}
+    >
+      Hello world
+    </div>
+  );
+
+  t.equal(isVisible(element), false);
+});
+
+test("isVisible() returns false for a text node with no data", (t) => {
+  const text = h.text("");
+
+  t.equal(isVisible(text), false);
+});
+
+test("isVisible() returns false for a text node with no data", (t) => {
+  const text = h.text("");
+
+  t.equal(isVisible(text), false);
+});
+
+test("isVisible() returns false for a text node with only whitespace", (t) => {
+  const text = h.text(" ");
+
+  t.equal(isVisible(text), false);
+});
+
+test("isVisible() returns false for a text node with a transparent color", (t) => {
```

Comment:
For this and the following examples, I'm not sure the `<div>` element gets the same result as its text 🤔

Reply:
Very good point 👍 I've adjusted the test cases.